apps/s_client.c

```diff
@@ -798,6 +801,31 @@ static int new_session_cb(SSL *S, SSL_SESSION *sess)
     return 0;
 }
 
+static void keylog_callback(const SSL *ssl, const char *line)
+{
+    BIO *keylog = BIO_new_file(keylog_file, "a");
+    char buffer[256];
```

Comment:
By the way, what do you think about the suggestion to always provide a newline in the callback? The tests in the tree also have a newline and I expect that every user will also append this newline.